packages/core/src/metrics/aggregator.ts

```diff
@@ -62,7 +63,11 @@ export class MetricsAggregator implements MetricsAggregatorBase {
     const tags = sanitizeTags(unsanitizedTags);
 
     const bucketKey = getBucketKey(metricType, name, unit, tags);
+
     let bucketItem = this._buckets.get(bucketKey);
+    // If this is a set metric, we need to calculate the delta from the previous weight.
+    const previousWeight = bucketItem && metricType === 's' ? bucketItem.metric.weight : 0;
```

Comment:
you should be able to use `SET_METRIC_TYPE` from constants instead of `'s'` here.